lib/fastly_nsq/message_queue/listener.rb

```diff
@@ -1,8 +1,10 @@
 module MessageQueue
   class Listener
-    def initialize(topic:, channel:)
-      @topic = topic
-      @channel = channel
+    def initialize(topic:, channel:, processor: nil, consumer: nil)
+      @topic     = topic
+      @channel   = channel
+      @processor = processor || DEFAULT_PROCESSOR
```

Comment:
I prefer the lack of line feeds personally because I like controlling
vertical space. I'll happily follow whatever convention people feel most
strongly about though.

On Mon, May 9, 2016 at 10:04 AM Adarsh Pandit <notifications@github.com>
wrote:

> In lib/fastly_nsq/message_queue/listener.rb
> <https://github.com/fastly/fastly_nsq/pull/26#discussion_r62534456>:
>
> > @@ -1,8 +1,10 @@
> >  module MessageQueue
> >    class Listener
> > -    def initialize(topic:, channel:)
> > -      @topic = topic
> > -      @channel = channel
> > +    def initialize(topic:, channel:, processor: nil, consumer: nil)
> > +      @topic     = topic
> > +      @channel   = channel
> > +      @processor = processor || DEFAULT_PROCESSOR
>
> Right. Actually, on re-reading, your implementation is clearer,
> particularly because it's broken out into a separate line. This is
> particularly true with the next assignment down, which is more involved.
> Please ignore my comment then.
> … <#m_-1839907133268486504_>
> On Mon, May 9, 2016 at 9:56 AM, Joshua Wehner ***@***.***> wrote: In
> lib/fastly_nsq/message_queue/listener.rb <#26 (comment)
> <https://github.com/fastly/fastly_nsq/pull/26#discussion_r62533183>>: >
> @@ -1,8 +1,10 @@ > module MessageQueue > class Listener > - def
> initialize(topic:, channel:) > - @topic <https://github.com/topic> =
> topic > - @channel <https://github.com/channel> = channel > + def
> initialize(topic:, channel:, processor: nil, consumer: nil) > + @topic
> <https://github.com/topic> = topic > + @channel
> <https://github.com/channel> = channel > + @processor
> <https://github.com/processor> = processor || DEFAULT_PROCESSOR That's
> true. Is it more readable/intention-revealing in that form? I think,
> long-term, I'd like to see us move away from the "magic constant"
> MessageProcessor in favor of this kind of configuration (maybe with some
> sane defaults), so if we need to establish how we're going to write that
> sort of thing now, maybe it's a good time. — You are receiving this because
> you were mentioned. Reply to this email directly or view it on GitHub <
> https://github.com/fastly/fastly_nsq/pull/26/files/01186203103b8ccd1e5e8ed9fe4914e0bc234ea1#r62533183>
>
> -- Adarsh Pandit External Consultant +1 313 454 1515 (m) @adarshp
> <https://github.com/adarshp> <https://twitter.com/adarshp>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/fastly/fastly_nsq/pull/26/files/01186203103b8ccd1e5e8ed9fe4914e0bc234ea1#r62534456>
>